modules/library/http/src/main/java/org/geotools/http/AbstractHttpClient.java

```diff
@@ -90,6 +108,33 @@ public boolean isTryGzip() {
         return tryGzip;
     }
 
+    protected static URL appendURL(URL oldUrl, Map<String, Object> appendQuery)
+            throws MalformedURLException {
+        String oldQuery = oldUrl.getQuery();
+
+        StringJoiner stringJoiner = new StringJoiner("&");
+        appendQuery.forEach(
+                (key, value) -> {
+                    try {
+                        stringJoiner.add(
+                                URLEncoder.encode(key, "UTF-8")
+                                        + "="
+                                        + URLEncoder.encode(value.toString(), "UTF-8"));
```

Comment:
What kind of content is expected here as values? We do have the GeoTools converters API that can handle information with more of a text focus then `toString()` method.

Reply:
What is expected if oldQuery has some of the same keys?

Reply:
The authkey value could have been a number, but there was no reason for it, so I changed it from Map<String, Object> to Map<String, String>, thus avoiding .toString()

Duplicate keys are now tested in https://github.com/geotools/geotools/compare/main...petersmythe:geotools:auth-for-cascaded-stores#diff-6ac812206449a240da095b61e785458b9c9837991a0e4531afafe37c1f46dffdR142 and work as expected (i.e. both values are present)